js/adapt-contrib-bookmarking.js

```diff
@@ -109,6 +170,19 @@ class Bookmarking extends Backbone.Controller {
     this.stopListening(Adapt, 'bookmarking:cancel');
   }
 
+  navigateToFurthest() {
+    const furthestModel = this.furthestIncompleteModel;
+    const furthestId = furthestModel.get('_id');
```

Comment:
If all of the models are `_isComplete = true` I expect that `furthestIncompleteModel` will return `undefined` causing line 175 to produce a `cannot get on undefined` error. This highlights that the use-case where someone clicks `Resume` when a course is complete will need some more work.

Perhaps the button should be disabled if there's no furthest or previous place to resume to? And/or the resume id calculation should change over to `"previous"` if `"furthest"` isn't productive?

Reply:
Yes, we did accommodate for this before by removing the button altogether, but that was project-specific. What about this order? If no incomplete models exist:
1. Use the previous set location.
2. If previously set location is the menu, remove the instance of the button. This would only be in play if the button is placed on the menu.

Reply:
Cool. 

Can we make it disabled instead of removing it? Removing UI is probably not good general accessibility or design practice. 

The previous location will still change when navigating a completed revisited course, I don't see how the previous location would usually or always be at the resume button. Either way the button should probably be disabled if it is to resume at itself. Hmm..

Reply:
Addressed!